lib/testers/rule-tester.js

```diff
@@ -296,9 +285,11 @@ RuleTester.prototype = {
             } else {
                 code = item.code;
 
-                // Assumes everything on the item is a config except for the
-                // parameters used by this tester
-                const itemConfig = lodash.omit(item, RuleTesterParameters);
+                // Validate only config properties
+                const schemaProperties = Object.keys(configSchema.properties);
+
+                // const schemaValidator = validator(configSchema, { verbose: true });
+                const itemConfig = lodash.pick(item, schemaProperties);
```

Comment:
It would be nice indeed, I was hoping to do that. Unfortunately we are using the test object as the config too and some tests add additional properties for their internal use, so they fail (I can look for some examples if you want). We could fix those tests, but plugins could be doing the same thing, so it seemed safer to not restrict it.